sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/CharsetProvider.scala

```diff
@@ -24,7 +24,7 @@
 private[sql] object CharsetProvider {
 
   final lazy val VALID_CHARSETS =
-    Set("us-ascii", "iso-8859-1", "utf-8", "utf-16be", "utf-16le", "utf-16", "utf-32")
+    Array("us-ascii", "iso-8859-1", "utf-8", "utf-16be", "utf-16le", "utf-16", "utf-32").sorted
```

Comment:
Is it more performant to use the contains of set than array?

Reply:
It's happening once anyway so shouldn't be a biggie. `invalidCharsetError` happens for every call

Reply:
charset can be a column value，this could be expensive？

Reply:
Hm, in that case, we should probably keep this as a set, and just create another variable sorted.

Reply:
https://www.scala-lang.org/api/2.12.5/scala/collection/SortedSet$.html maybe we can use a SortedSet here

Reply:
Great idea!

Reply:
Here: https://github.com/apache/spark/pull/48245

Reply:
Sounds good!